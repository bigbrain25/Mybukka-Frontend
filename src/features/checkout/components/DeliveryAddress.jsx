import React, { Fragment, useState, useEffect } from 'react';

import TextArea from 'Components/input/TextArea';
import PropTypes, { any } from 'prop-types';
import DeliveryOrPickupNav from 'Components/common-navs/DeliveryOrPickupNav';
import Button from 'Components/button/Button';
import inputField from '../InputAttribute/inputData.json';
import { validateAField, validateAllFields } from '../validation/validateField';

import './payment.scss';
import './deliveryAddress.scss';
import AuthForm from '../common/AuthForm';
import Demarcation from '../common/SmallScreenDivider';

const DeliveryForm = ({
  setWrapperRef,
  inputData,
  validationErrors,
  handleSaveButton,
  autoComplete,
  handleChange
}) => (
  <form ref={setWrapperRef} className="border padding-20 mt-4">
    <AuthForm
      inputData={inputData}
      inputField={inputField.deliveryAddress}
      handleChange={handleChange}
      errors={validationErrors}
      autoComplete={autoComplete}
    />
    <div className="form-group mb-4">
      <TextArea
        placeholderText="Add delivery instructuctions..."
        name="deliveryInstruction"
        classNames="instruction"
      />
    </div>
    <div>
      <Button
        type="button"
        text="Save"
        classNames="small-button-save"
        handleClick={handleSaveButton}
      />
    </div>
  </form>
);

const Pickup = ({ title, name }) => (
  <section className="container mb-2 mt-4">
    <h2 className="font-size-16">{title}</h2>
    <ul className="list-group mt-4 time">
      <li className="list-group-item">{name}</li>
    </ul>
  </section>
);

const Delivery = () => {
  let wrapperRef;
  const [isPickup, setIsPickup] = useState(false);
  const [autoComplete, setAutoComplete] = useState(false);
  const [validationErrors, setValidationErrors] = useState({
    streetAddress1: '',
    streetAddress2: '',
    contactName: '',
    phoneNumber: ''
  });

  const [inputData, setInputData] = useState({
    streetAddress1: '',
    streetAddress2: '',
    contactName: '',
    phoneNumber: ''
  });

  const setWrapperRef = (node) => {
    wrapperRef = node;
  };

  const handleClickOutside = (event) => {
    if (wrapperRef && !wrapperRef.contains(event.target)) {
      setAutoComplete(false);
    }
  };

  useEffect(() => {
    document.addEventListener('mousedown', handleClickOutside);
  });

  const handleChange = ({ target: { name, value } }) => {
    const newFieldData = { [name]: value };
    const validation = validateAField(newFieldData, name);
    setInputData({
      ...inputData,
      ...newFieldData
    });
    setValidationErrors({
      ...validationErrors,
      [name]: validation.message
    });
    if (name === 'streetAddress1') {
      setAutoComplete(true);
    }
  };

  const handleSaveButton = (e) => {
    e.preventDefault();
    const validation = validateAllFields(inputData);
    setValidationErrors({
      ...validationErrors,
      ...validation
    });
  };

  return (
    <div className="mb-2 mt-4">
      <h1 className="font-size-36 px-3 px-md-3 px-lg-0">Checkout</h1>
      <div className="col-md-12 col-lg-6 p-0 mb-4 mt-4 height-50">
        <DeliveryOrPickupNav handleClick={() => setIsPickup(!isPickup)} />
      </div>
      <Demarcation />
      <h2 className="font-size-16 px-3 px-md-3 px-lg-0">Delivery Address</h2>
      <DeliveryForm
        inputData={inputData}
        handleChange={handleChange}
        validationErrors={validationErrors}
        autoComplete={autoComplete}
        setWrapperRef={setWrapperRef}
        handleSaveButton={handleSaveButton}
      />
      {isPickup && (
        <Fragment>
          <Pickup
            title="Pickup Address"
            name="801 Mission St., San Francisco (1.0 mi)"
          />
          <Pickup title="Pickup time" name="Ready in 20 min" />
        </Fragment>
      )}
    </div>
  );
};

export default Delivery;

DeliveryForm.propTypes = {
  inputData: PropTypes.objectOf(any).isRequired,
  handleChange: PropTypes.func.isRequired,
  autoComplete: PropTypes.bool.isRequired,
  setWrapperRef: PropTypes.objectOf(any).isRequired,
  handleSaveButton: PropTypes.func.isRequired,
  validationErrors: PropTypes.objectOf(any).isRequired
};

Pickup.propTypes = {
  title: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired
};
