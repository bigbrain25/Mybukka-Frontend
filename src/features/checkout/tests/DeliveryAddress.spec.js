import React from 'react';

import DeliveryAddress from '../components/DeliveryAddress';

describe('Area to explore component', () => {
  const props = {
    handleClick: jest.fn(),
    handleChange: jest.fn()
  };
  const wrapper = mount(<DeliveryAddress {...props} />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
