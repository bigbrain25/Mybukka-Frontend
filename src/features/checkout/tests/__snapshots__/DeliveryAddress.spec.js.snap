// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Area to explore component renders properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mb-2 mt-4"
      >
        <h1
          class="font-size-36 px-3 px-md-3 px-lg-0"
        >
          Checkout
        </h1>
        <div
          class="col-md-12 col-lg-6 p-0 mb-4 mt-4 height-50"
        >
          <div
            class="btn-group"
            role="group"
          >
            <button
              class="btn-delivery "
              type="button"
            >
              delivery
            </button>
            <button
              class="btn-pickup delivery-mode-active"
              type="button"
            >
              pickup
            </button>
          </div>
        </div>
        <div
          class="demarcation"
        />
        <h2
          class="font-size-16 px-3 px-md-3 px-lg-0"
        >
          Delivery Address
        </h2>
        <form
          class="border padding-20 mt-4"
        >
          <div
            class="form-group font-size-14 mb-4"
          >
            <label
              class="no-label"
              for="address"
            >
              Address
            </label>
            <input
              class="form-control border-input-style p-0"
              name="streetAddress1"
              placeholder="Address"
              type="text"
              value=""
            />
            <span
              class="text-danger font-size-11"
            />
          </div>
          <div
            class="form-group font-size-14 mb-4"
          >
            <label
              class="no-label"
              for="streetAddress2"
            >
              Apt no or company name
            </label>
            <input
              class="form-control border-input-style p-0"
              name="streetAddress2"
              placeholder="Apt no or company name"
              type="text"
              value=""
            />
            <span
              class="text-danger font-size-11"
            />
          </div>
          <div
            class="d-inline-flex pl-0 flex-column col-6 col-md-6 mb-4"
          >
            <label
              class="no-label"
              for="contactName"
            >
              Contact name
            </label>
            <input
              class="form-control border-input-style p-0"
              name="contactName"
              placeholder="Contact name"
              type="text"
              value=""
            />
            <span
              class="text-danger font-size-11"
            />
          </div>
          <div
            class="d-inline-flex pr-0 flex-column col-6 col-md-6 mb-4"
          >
            <label
              class="no-label"
              for="phoneNumber"
            >
              Phone number
            </label>
            <input
              class="form-control border-input-style p-0"
              name="phoneNumber"
              placeholder="Phone number"
              type="text"
              value=""
            />
            <span
              class="text-danger font-size-11"
            />
          </div>
          <div
            class="form-group mb-4"
          >
            <textarea
              class="text-area instruction"
              maxlength="200"
              name="deliveryInstruction"
              placeholder="Add delivery instructuctions..."
            />
          </div>
          <div>
            <button
              class="small-button-save"
              data-target=""
              data-testid="button"
              data-toggle=""
              id="n8J8mD2FM"
              type="button"
            >
              Save
            </button>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mb-2 mt-4"
    >
      <h1
        class="font-size-36 px-3 px-md-3 px-lg-0"
      >
        Checkout
      </h1>
      <div
        class="col-md-12 col-lg-6 p-0 mb-4 mt-4 height-50"
      >
        <div
          class="btn-group"
          role="group"
        >
          <button
            class="btn-delivery "
            type="button"
          >
            delivery
          </button>
          <button
            class="btn-pickup delivery-mode-active"
            type="button"
          >
            pickup
          </button>
        </div>
      </div>
      <div
        class="demarcation"
      />
      <h2
        class="font-size-16 px-3 px-md-3 px-lg-0"
      >
        Delivery Address
      </h2>
      <form
        class="border padding-20 mt-4"
      >
        <div
          class="form-group font-size-14 mb-4"
        >
          <label
            class="no-label"
            for="address"
          >
            Address
          </label>
          <input
            class="form-control border-input-style p-0"
            name="streetAddress1"
            placeholder="Address"
            type="text"
            value=""
          />
          <span
            class="text-danger font-size-11"
          />
        </div>
        <div
          class="form-group font-size-14 mb-4"
        >
          <label
            class="no-label"
            for="streetAddress2"
          >
            Apt no or company name
          </label>
          <input
            class="form-control border-input-style p-0"
            name="streetAddress2"
            placeholder="Apt no or company name"
            type="text"
            value=""
          />
          <span
            class="text-danger font-size-11"
          />
        </div>
        <div
          class="d-inline-flex pl-0 flex-column col-6 col-md-6 mb-4"
        >
          <label
            class="no-label"
            for="contactName"
          >
            Contact name
          </label>
          <input
            class="form-control border-input-style p-0"
            name="contactName"
            placeholder="Contact name"
            type="text"
            value=""
          />
          <span
            class="text-danger font-size-11"
          />
        </div>
        <div
          class="d-inline-flex pr-0 flex-column col-6 col-md-6 mb-4"
        >
          <label
            class="no-label"
            for="phoneNumber"
          >
            Phone number
          </label>
          <input
            class="form-control border-input-style p-0"
            name="phoneNumber"
            placeholder="Phone number"
            type="text"
            value=""
          />
          <span
            class="text-danger font-size-11"
          />
        </div>
        <div
          class="form-group mb-4"
        >
          <textarea
            class="text-area instruction"
            maxlength="200"
            name="deliveryInstruction"
            placeholder="Add delivery instructuctions..."
          />
        </div>
        <div>
          <button
            class="small-button-save"
            data-target=""
            data-testid="button"
            data-toggle=""
            id="n8J8mD2FM"
            type="button"
          >
            Save
          </button>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
